### PR TITLE
BUG: DataFrameBinOp tile output dtype is wrong

### LINKED ISSUE #362: BUG: unsupported operand type(s) for &: 'DatetimeArray' and 'DatetimeArray' in case of GPU
Note that the issue tracker is NOT the place for general support.  For
discussions about development, questions about usage, or any general questions,
contact us on https://discuss.xorbits.io/.
GPU TPCH Q20:
```
Traceback (most recent call last):
  File "/home/lichengjie/workspace/xorbits/benchmarks/tpch/run_queries.py", line 1141, in <module>
    main()
  File "/home/lichengjie/workspace/xorbits/benchmarks/tpch/run_queries.py", line 1128, in main
    run_queries(
  File "/home/lichengjie/workspace/xorbits/benchmarks/tpch/run_queries.py", line 1058, in run_queries
    globals()[f"q{query:02}"](*queries_to_args[query])
  File "/home/lichengjie/workspace/xorbits/benchmarks/tpch/run_queries.py", line 122, in wrapped
    q(*args, **kwargs)
  File "/home/lichengjie/workspace/xorbits/benchmarks/tpch/run_queries.py", line 930, in q20
    print(total)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/utils.py", line 37, in inn
    return f(self, *args, **kwargs)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/core/data.py", line 261, in __str__
    run(self)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/core/execution.py", line 55, in run
    mars_execute(mars_tileables, **kwargs)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/deploy/oscar/session.py", line 1877, in execute
    return session.execute(
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/deploy/oscar/session.py", line 1671, in execute
    execution_info: ExecutionInfo = fut.result(
  File "/home/lichengjie/miniconda3/lib/python3.9/concurrent/futures/_base.py", line 446, in result
    return self.__get_result()
  File "/home/lichengjie/miniconda3/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/deploy/oscar/session.py", line 1857, in _execute
    await execution_info
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/deploy/oscar/session.py", line 104, in wait
    return await self._aio_task
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/deploy/oscar/session.py", line 952, in _run_in_background
    raise task_result.error.with_traceback(task_result.traceback)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/services/task/supervisor/processor.py", line 387, in run
    async for stage_args in self._iter_stage_chunk_graph():
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/services/task/supervisor/processor.py", line 171, in _iter_stage_chunk_graph
    chunk_graph = await self._get_next_chunk_graph(chunk_graph_iter)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/services/task/supervisor/processor.py", line 162, in _get_next_chunk_graph
    chunk_graph = await fut
  File "/home/lichengjie/miniconda3/lib/python3.9/asyncio/threads.py", line 25, in to_thread
    return await loop.run_in_executor(None, func_call)
  File "/home/lichengjie/miniconda3/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/services/task/supervisor/processor.py", line 157, in next_chunk_graph
    return next(chunk_graph_iter)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/services/task/supervisor/preprocessor.py", line 201, in tile
    for chunk_graph in chunk_graph_builder.build():
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/core/graph/builder/chunk.py", line 431, in build
    yield from self._build()
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/core/graph/builder/chunk.py", line 425, in _build
    graph = next(tile_iterator)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/services/task/supervisor/preprocessor.py", line 75, in _iter_without_check
    to_update_tileables = self._iter()
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/core/graph/builder/chunk.py", line 308, in _iter
    self._tile(
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/core/graph/builder/chunk.py", line 201, in _tile
    need_process = next(tile_handler)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/core/graph/builder/chunk.py", line 173, in _tile_handler
    tiled_tileables = yield from handler.tile(tiled_tileables)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/core/entity/tileables.py", line 80, in tile
    tiled_result = yield from tile_handler(op)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/dataframe/arithmetic/core.py", line 358, in tile
    return cls._tile_both_series(op)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/dataframe/arithmetic/core.py", line 100, in _tile_both_series
    op.copy()
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/core/operand/core.py", line 174, in new_chunk
    return self.new_chunks(inputs, kws=kws, **kw)[0]
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/core/operand/core.py", line 166, in new_chunks
    return self._new_chunks(inputs, kws=kws, **kwargs)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/dataframe/arithmetic/core.py", line 627, in _new_chunks
    properties = self._calc_properties(*property_inputs)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/dataframe/arithmetic/core.py", line 582, in _calc_properties
    dtype = infer_dtype(x1.dtype, x2.dtype, cls._operator)
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/dataframe/utils.py", line 945, in infer_dtype
    return operator(left, right).dtype
  File "/home/lichengjie/workspace/xorbits/python/xorbits/_mars/dataframe/arithmetic/__init__.py", line 127, in wrapped
    return raw_method(self, other)
  File "/home/lichengjie/miniconda3/lib/python3.9/site-packages/pandas/core/ops/common.py", line 72, in new_method
    return method(self, other)
  File "/home/lichengjie/miniconda3/lib/python3.9/site-packages/pandas/core/arraylike.py", line 72, in __and__
    return self._logical_method(other, operator.and_)
  File "/home/lichengjie/miniconda3/lib/python3.9/site-packages/pandas/core/series.py", line 6254, in _logical_method
    res_values = ops.logical_op(lvalues, rvalues, op)
  File "/home/lichengjie/miniconda3/lib/python3.9/site-packages/pandas/core/ops/array_ops.py", line 380, in logical_op
    res_values = op(lvalues, rvalues)
TypeError: unsupported operand type(s) for &: 'DatetimeArray' and 'DatetimeArray'
```

Minimum reproduce:
```python
date1 = md.Timestamp('1996-01-01')
date2 = md.Timestamp('2005-01-01')
a = md.DataFrame({'date': [datetime.datetime(1999, 1, 1), datetime.datetime(2000, 1, 1)], 'col': [1, 2]}).to_gpu()
res = (a.date >= date1) & (a.date < date2)
print(res.execute())
```

### LINKED ISSUE #365: BUG: DataFrameBinOp output dtype error during tile
Note that the issue tracker is NOT the place for general support.  For
discussions about development, questions about usage, or any general questions,
contact us on https://discuss.xorbits.io/.

During DataFrameBinOp tile, the output dtype not match with the actual output.

Reproduce:
Comment out @register_operand_based_optimization_rule([DataFrameUnaryUfunc, DataFrameBinopUfunc])

Then execute:
```python
import datetime
date1 = md.Timestamp('1996-01-01')

a = md.DataFrame({'date': [datetime.datetime(1999, 1, 1), datetime.datetime(2000, 1, 1)], 'col': [1, 2]})
res = a.date >= date1
print(res.execute())
```

### PR DESCRIPTION
**Pull request by ChengjieLi28**
## Related issue number

Fixes #365 
Fixes #362 

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

File: python/xorbits/_mars/dataframe/arithmetic/tests/test_arithmetic_execution.py [hunk 1]
```diff
@@ -904,6 +906,80 @@ def test_date_time_bin(setup):
...
+                for bin_op in ("and", "or", "not", "xor"):
+                    yield data_type, compare_target_type, chunked, bin_op
+
+
+@require_cudf
+@pytest.mark.parametrize(
+    "data_type,compare_target_type,chunked,bin_op",
+    _generate_params_for_gpu(),
+)
+def test_date_time_bin_gpu(data_type, compare_target_type, chunked, bin_op, setup_gpu):
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/arithmetic/tests/test_arithmetic_execution.py:922]: As this PR works for both GPU and CPU, we can use `support_cuda` annotation introduced in #368, waiting that PR be merged.

Comment (ChengjieLi28) [reply python/xorbits/_mars/dataframe/arithmetic/tests/test_arithmetic_execution.py]: Added.

**Review by aresnow1 — APPROVED** [review]
LGTM

**Review by qianduoduo0904 — APPROVED** [review]
LGTM

**Pull request merged**